ElDorito/Source/Modules/ModuleGame.cpp

```diff
@@ -1056,6 +1063,57 @@ namespace
 
 	}
 
+	bool CommandDownloadVariant(const std::vector<std::string>& Arguments, std::string& returnInfo)
+	{
+		if (Arguments.size() < 1)
+			return false;
+		returnInfo = "Trying to download:" + Arguments[0] + "\n";
+		HttpRequest req(L"ElDewrito/" + Utils::String::WidenString(Utils::Version::GetVersionString()), L"", L"");
+		if (!req.SendRequest(Utils::String::WidenString(Arguments[0]), L"GET", L"", L"", L"", NULL, 0))
+		{
+			returnInfo += "Error fetching variant file. Error code: " + std::to_string(req.lastError);
+			return false;
+		}
+
+		// mapv = Map Variant, mpvr = Game Mode (Multiplayer) Variant
+		std::string variantFormat = std::string(&req.responseBody.at(312), &req.responseBody.at(316));
+		returnInfo += "Variant Format Header (" + variantFormat + ")\n";
+
+		if (req.responseBody.size() != 0x1000 && req.responseBody.size() != 0xF000)
+		{
+			returnInfo += "Incorrect variant file size (" + std::to_string(req.responseBody.size()) + " bytes)";
+			return false;
+		}
+		returnInfo += "Response variant file size (" + std::to_string(req.responseBody.size()) + " bytes)\n";
+		std::string header = std::string(&req.responseBody.at(0x0), &req.responseBody.at(0x4));
+		bool valid = (header == "_blf");
+
+		BYTE variant_name[0x20];
+		// Default entire array to 0
+		std::fill_n(variant_name, 0x20, 0);
+
+		for (int i = 0; i <= 31; i++)
+			variant_name[i] = req.responseBody[i + 72];
```

Comment:
can't `strcpy_s` be used here?

Reply:
I've attempted to use strcpy_s however I needed to first then convert req.responseBody into a char* which was causing issues with all the nulls in it.
(this part was also based off your original attempt of supporting map downloading in the console)

Reply:
use `wcscpy_s` then.

Reply:
why not contribute directly to his fork on his git @ https://github.com/thiconz/ElDorito

I'll merge the PR, and make sure it stays updated with no conflicts through 0.6.x